### PR TITLE
Updated Event List Cell Layout

### PR DESCRIPTION
**Pull request by avihut**
# New Event Cell Design

screen shot 2015-04-26 at 7 14 45 pm

This design is fully configured with auto layout, but without size classes. I might add support for different size classes later on if I have the time. Currently this layout is configured for Any / Any.

The design guide was the Facebook event list item as shown in your Events page:

screen_shot_2015-04-22_at_11 12 41_am 1

The thumbnail picture ended up a little more squashed to make head room for the title, and to make a little more room on the right for the payment setup button. If I get to work on bigger size classes I might make it look a bit more like the original.
# Placeholder Images

Placeholder images will load before the events' actual cover images have finished being fetched asynchronously. They will also be used for events that don't have a configured cover image.

new events list

### CODE REVIEW

File: Event.swift [hunk 1]
```diff
@@ -36,7 +36,37 @@ class Event: PFObject, PFSubclassing {
...
+    
+    var totalMoneyToCollect: Int? {
+        return nil
+    }
+    
+}
+
+func <(lhs: Event, rhs: Event) -> Bool {
+    if let lhsStartTime = lhs.startTime {
+        if let rhsStartTime = rhs.startTime {
```
Comment (yonbergman) [inline Event.swift]: if let x, let y {

File: Event.swift [hunk 2]
```diff
@@ -36,7 +36,37 @@ class Event: PFObject, PFSubclassing {
...
+    @NSManaged var endTime: NSDate?
+    var coverImageUrl: NSURL?
     
+    var moneyCollected: Int? {
+        return nil
+    }
+    
+    var totalMoneyToCollect: Int? {
+        return nil
+    }
```
Comment (yonbergman) [inline Event.swift:48]: why the computed properties here?

File: PSix/EventCell.swift [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    @IBOutlet weak var eventDateMonthLabel: UILabel!
+    @IBOutlet weak var eventDateDayLabel: UILabel!
+    @IBOutlet weak var eventThumbImage: UIImageView!
+    @IBOutlet weak var paymentCollectionStatus: UILabel!
+    
+    private static var placeholderThumb = UIImage(named: "PlaceholderEventThumb")
+    
+    override func awakeFromNib() {
+        eventNameLabel.adjustsFontSizeToFitWidth = false
+        eventNameLabel.lineBreakMode = NSLineBreakMode.ByTruncatingTail
```
Comment (yonbergman) [inline PSix/EventCell.swift]: can be configured in storyboard

File: PSix/EventCell.swift [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            eventDateDayLabel.text = event?.startTime?.dayDoubleDigit
+            setDayAndTimeLabel()
+            setPaymentStatus()
+            setEventThumbImage()
+        }
+    }
+    
+    private func setDayAndTimeLabel() {
+        if let startDate = event?.startTime {
+            let dateFormatter = NSDateFormatter()
```
Comment (yonbergman) [inline PSix/EventCell.swift]: you can re-use this formatter and probably should.
I would extract all this logic to a presenter class and have the formatter static-ly shared

File: PSix/EventCell.swift [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            dateFormatter.dateFormat = "EEEE, h:mm"
+            let extendedText = dateFormatter.stringFromDate(startDate) as NSString
+            let extTextSize = extendedText.sizeWithAttributes([NSFontAttributeName: eventDateDayAndTime.font])
+            if eventDateDayAndTime.bounds.width < extTextSize.width {
```
Comment (yonbergman) [inline PSix/EventCell.swift]: 3 lines that should be extracted to something like - isTextWiderThanWidth:forFont

File: PSix/EventCell.swift [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        if let coverImageUrl = event?.coverImageUrl {
+            dispatch_async(dispatch_get_main_queue()) { [unowned self] () -> Void in
+                if let coverImageData = NSData(contentsOfURL: coverImageUrl) {
+                    self.eventThumbImage.image = UIImage(data: coverImageData)
+                }
+            }
+        } else {
+            eventThumbImage.image = EventCell.placeholderThumb
+        }
+    }
```
Comment (yonbergman) [inline PSix/EventCell.swift]: you have a caching issue here.
if i scroll the table and re-use cells two probe will arise
1. placeholder won't be set
2. you load the image multiple times from the net

File: PSix/FacebookService.swift [hunk 7]
```diff
@@ -34,6 +36,39 @@ class FacebookService {
         return Int(NSDate.timeIntervalSinceReferenceDate() + NSTimeIntervalSince1970)
     }
     
+    private static func extractLocation(eventData: AnyObject) -> String? {
+        if let locationData: AnyObject? = eventData[FBReq.Event.LOCATION] {
+            if let locationName = locationData?["name"] as? String {
```
Comment (yonbergman) [inline PSix/FacebookService.swift]: you can use multiple lets in the same if, even using previous results

``` swift
if let foo = something(), bar = foo.somethingElse
```

**Comment by yonbergman** [comment]
# Storyboard  comments

screen shot 2015-04-27 at 9 54 19 am
Try to never have negative constraints - this usually means that you have the wrong ones set up (in this case I think you probable have a problem of anchors)

screen shot 2015-04-27 at 9 54 24 am
You have multiple constraints here where you could have found a way to represent this relationship with only one constraint. (either wrap them all in a view, or have one of them have a vertical spacing constraint and the rest align left to him)

File: Event.swift (hunk 1 above)
Comment (avihut) [inline Event.swift]: Fixed.

I initially wanted to divide those `if`s to better control intermediate state for comparison, but it came out not so readable, so I refactored it.

File: Event.swift (hunk 2 above)
Comment (avihut) [inline Event.swift:48]: It's a preparation for the payment collection mechanism. I'm going to implement this further in the coming ticket.

File: PSix/EventCell.swift (hunk 3 above)
Comment (avihut) [inline PSix/EventCell.swift]: It's a leftover from an intermediate state. Removed it.

File: PSix/EventCell.swift (hunk 4 above)
Comment (avihut) [inline PSix/EventCell.swift]: Fixed

File: PSix/EventCell.swift (hunk 5 above)
Comment (avihut) [inline PSix/EventCell.swift]: Fixed

File: PSix/FacebookService.swift (hunk 7 above)
Comment (avihut) [inline PSix/FacebookService.swift]: Fixed

Comment (avihut) [inline PSix/FacebookService.swift]: Fixed

File: PSix/EventCell.swift (hunk 6 above)
Comment (avihut) [inline PSix/EventCell.swift]: Fixed this. Now the placeholder is set at the beginning, and the presenter takes care of fetching and caching event images.

**Pull request closed without merging**